source/boundary_traction/function.cc

```diff
@@ -42,10 +43,39 @@ namespace aspect
                        const Tensor<1,dim> &) const
     {
       Tensor<1,dim> traction;
-      for (unsigned int d=0; d<dim; ++d)
-        traction[d] = boundary_traction_function.value(p,d);
+      if (coordinate_system == ::aspect::Utilities::Coordinates::CoordinateSystem::cartesian)
+        {
+          for (unsigned int d=0; d<dim; ++d)
+            traction[d] = boundary_traction_function.value(p,d);
+
+          return traction;
+        }
+      else if (coordinate_system == ::aspect::Utilities::Coordinates::CoordinateSystem::spherical)
+        {
+          const std_cxx11::array<double,dim> spherical_coordinates =
+            aspect::Utilities::Coordinates::cartesian_to_spherical_coordinates(p);
+          Point<dim> point;
+
+          for (unsigned int d=0; d<dim; ++d)
+            point[d] = spherical_coordinates[d];
+          for (unsigned int d=0; d<dim; ++d)
+            traction[d] = boundary_traction_function.value(point,d);
+
+          return traction;
+        }
+      else
+        {
+          const double depth = this->get_geometry_model().depth(p);
+          Point<dim> point;
+          point(0) = depth;
+
+          for (unsigned int d=0; d<dim; ++d)
+            traction[d] = boundary_traction_function.value(point,d);
+
+          return traction;
+        }
+
 
-      return traction;
```

Comment:
lets leave the `return traction` here, and remove it from within the if-else clauses. That is not a strict rule, just my preference (there is a debate about which practice is better).